front/src/utils/__tests__/trainId.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { describe, it, expect } from 'vitest';
+
+import type { TrainScheduleId, OccurrenceId, PacedTrainId } from 'reducers/osrdconf/types';
+
+import {
+  formatEditoastTrainIdToTrainScheduleId,
+  formatEditoastTrainIdToOccurrenceId,
+  formatTrainScheduleIdToEditoastTrainId,
+  formatOccurrenceIdToEditoastTrainId,
+  formatEditoastTrainIdToPacedTrainId,
+  formatPacedTrainIdToEditoastTrainId,
+} from '../trainId';
+
+describe('formatEditoastTrainIdToTrainScheduleId', () => {
+  it('should format to a TrainScheduleId', () => {
+    const trainId = 123;
+    const result = formatEditoastTrainIdToTrainScheduleId(trainId);
+    expect(result).toEqual(`train-${trainId}`);
+  });
+});
+
+describe('formatEditoastTrainIdToPacedTrainId', () => {
+  it('should format to a TrainScheduleId', () => {
+    const trainId = 123;
+    const result = formatEditoastTrainIdToPacedTrainId(trainId);
+    expect(result).toEqual(`paced-${trainId}`);
+  });
+});
+
+describe('formatEditoastTrainIdToOccurrenceId', () => {
+  it('should format a valid paced train ID and occurrence index correctly', () => {
+    const pacedTrainId = 123;
+    const occurrenceIndex = 1;
+    const result = formatEditoastTrainIdToOccurrenceId({ pacedTrainId, occurrenceIndex });
+    expect(result).toBe(`paced-${pacedTrainId}-occurrence-${occurrenceIndex}`);
+  });
+});
+
+describe('formatTrainScheduleIdToEditoastTrainId', () => {
+  it('should return a valid editoast id', () => {
+    const trainScheduleId: TrainScheduleId = 'train-123' as TrainScheduleId;
```

Comment:
You don't need the 2 casts here :)
```suggestion
    const trainScheduleId = 'train-123' as TrainScheduleId;
```

It's the same in the other functions

Reply:
Oups, done